### ISSUE TITLE
[com.micropole.android.tcl_mobile] crashes with "java.lang.NullPointerException: Listener must not be null" when using geolocation

### ISSUE THREAD

**Issue by mossroy** [microg/GmsCore#1982]
**Affected app**
Name: TCL Mobile (public transportation in France)
Package id: com.micropole.android.tcl_mobile

**Describe the bug**
The app worked fine since a long time.
But, after upgrading /e/ OS to 1.12.3 instead of 1.11, it now crashes whenever using geolocation

**To Reproduce**
1. Install the app
2. Skip the intro screens ("Passer" in French on top right)
3. Click on "Start" ("Départ" in French)
4. Click on "My location" ("Ma position" in French)
5. Accept geolocation
6. The app crashes

**Expected behavior**
Geolocation should be found and used by the app, like it was before

**System**
Android Version: 11 and 12
Custom ROM: /e/ OS 1.12.3 (tested on two different FP3, and a SM-G900F)

**microG**
microG Core version: 0.2.28.231657-17 (9645c3e)-noen
microG Self-Check results: All ticked

**Additional context**
It was not crashing with /e/ OS 1.11

Adb log excerpt:
```
07-03 18:27:26.674  1665  1665 D AndroidRuntime: Shutting down VM
07-03 18:27:26.674  1665  1665 E AndroidRuntime: FATAL EXCEPTION: main
07-03 18:27:26.674  1665  1665 E AndroidRuntime: Process: com.micropole.android.tcl_mobile, PID: 1665
07-03 18:27:26.674  1665  1665 E AndroidRuntime: java.lang.NullPointerException: Listener must not be null
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.common.internal.Preconditions.checkNotNull(com.google.android.gms:play-services-basement@@18.0.0:2)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.common.api.internal.ListenerHolders.createListenerKey(com.google.android.gms:play-services-base@@18.0.1:1)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.location.FusedLocationProviderClient.removeLocationUpdates(Unknown Source:6)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.airbnb.android.react.maps.FusedLocationSource.deactivate(FusedLocationSource.java:70)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.maps.zzs.deactivate(com.google.android.gms:play-services-maps@@17.0.1:1)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.maps.internal.zzj.zza(com.google.android.gms:play-services-maps@@17.0.1:1)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.internal.maps.zzb.onTransact(com.google.android.gms:play-services-maps@@17.0.1:3)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at android.os.Binder.transact(Binder.java:1064)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.maps.internal.ILocationSourceDelegate$Stub$Proxy.deactivate(ILocationSourceDelegate.java:121)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at org.microg.gms.maps.mapbox.SourceLocationEngine.removeLocationUpdates(SourceLocationEngine.kt:41)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at org.microg.gms.maps.mapbox.GoogleMapImpl.updateLocationEngineListener(GoogleMap.kt:423)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at org.microg.gms.maps.mapbox.GoogleMapImpl.setLocationSource(GoogleMap.kt:436)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.maps.internal.IGoogleMapDelegate$Stub.onTransact(IGoogleMapDelegate.java:474)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at org.microg.gms.maps.mapbox.GoogleMapImpl.onTransact(GoogleMap.kt:876)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at android.os.Binder.transact(Binder.java:1064)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.internal.maps.zza.zzc(com.google.android.gms:play-services-maps@@17.0.1:2)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.maps.internal.zzg.setLocationSource(com.google.android.gms:play-services-maps@@17.0.1:3)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.google.android.gms.maps.GoogleMap.setLocationSource(com.google.android.gms:play-services-maps@@17.0.1:2)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.airbnb.android.react.maps.AirMapView$15.onHostResume(AirMapView.java:400)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.facebook.react.bridge.ReactContext$1.run(ReactContext.java:224)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:938)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7870)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
07-03 18:27:26.674  1665  1665 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```
Workaround:
Reinstall the app (or remove its data), then do not use geolocation.

Original report: https://community.e.foundation/t/french-app-tcl-crashes-when-using-geolocation-since-v1-12/50501

**Comment by derkommutator** [comment]
I'm experiencing the same behaviour in A Better Route Planner.

**Comment by mar-v-in** [comment]
This is the same as #1962.

**Comment by mossroy** [comment]
Seems to be solved with /e/os 1.13
